### DISCUSSION TITLE
How do I handle many to many relationships when creating a new record - can not find it in docs

### DISCUSSION THREAD

**Discussion by DeveloperOskar** [pocketbase/pocketbase#1409] [Q&A]
I have 3 tables

"foods"
"categories"
"foods_categories" (join table)

A food can have many categories and a category can be on many foods. 

The way I create a new food right now is this:

```
const food = await this.pocketBaseService.pb.collection('foods').create<Food>({
  protein, fat, carbs, kcal, name, unit, user: [userId]
});

await this.pocketBaseService.pb.collection('foods_categories').create({
  foods: food.id,
  categories,
});
```
This works but if the second request fails, there is a risk of a food being created on without any categories. 

Is there a way to make this in one transaction? My guess is that I should use expand somehow but I can't find any food examples 

Thanks

**Comment by ganigeorgiev** [comment]
> Is there a way to make this in one transaction?

Unfortunately No. Bulk transactional create/update/delete is in the roadmap and eventually will be handled together with #48.

> My guess is that I should use expand somehow but I can't find any food examples

`expand` is only for retrieving data (if you haven't already, you can check https://pocketbase.io/docs/expanding-relations/).

---

Another alternative way to handle one-many relationships (one food - many categories) could be use a **multi-**`relation` type field (eg. Max select != 1) in the `foods` collection pointing to the categories. this way you'll pass the category id together with the main `foods` record during create.

But there are 2 limitations at the moment with the multi-relational fields:

- There is no multi-match filter support yet and when attempting to filter a multi-value field it will return `true` on the first match (eg. `someRelField.status = 'active'` will return true if there is at least one `someRelField` record that match the condition).
- When submitting/removing items you have to always pass the final array, eg:
   ```js
   await pb.collection('foods').update('RECORD_ID', { categories: ['id1', 'id2', 'id3', 'id4'] })
   ```

Both limitations will be fixed in the upcoming v0.11 release (_most likely sometime next week_).
